test/test_c10d.py

```diff
@@ -2886,30 +2900,25 @@ def _test_nccl_errors_blocking(self, func):
 
     @requires_nccl()
     @skip_if_not_multigpu
+    @per_process_setup
     def test_nccl_errors_blocking_clean_exit(self):
         self._test_nccl_errors_blocking(lambda : sys.exit(0))
 
     @requires_nccl()
     @skip_if_not_multigpu
+    @per_process_setup
     def test_nccl_errors_blocking_abort(self):
         self._test_nccl_errors_blocking(lambda : os.abort())
 
     @requires_nccl()
     @skip_if_not_multigpu
+    @per_process_setup
     def test_nccl_errors_blocking_sigkill(self):
         self._test_nccl_errors_blocking(lambda : os.kill(os.getpid(), signal.SIGKILL))
 
     @requires_nccl()
     @skip_if_not_multigpu
-    def test_nccl_errors_blocking_sigstop(self):
```

Comment:
any reason for removing this test?

Reply:
`SIGSTOP` explicitly tests for the timeout condition though, whereas the others test termination.

What about sharing the PIDs through an allgather?